lib/core/rlocks.nim

```diff
@@ -17,36 +17,50 @@ when not compileOption("threads") and not defined(nimdoc):
     {.error: "Rlocks requires --threads:on option.".}
 
 const insideRLocksModule = true
+const useOrcArc = defined(gcArc) or defined(gcOrc)
+
 include "system/syslocks"
 
 type
-  RLock* = SysLock ## Nim lock, re-entrant
+  RLock* = object
+    ## Nim lock, re-entrant
+    lock: SysLock
 
 proc initRLock*(lock: var RLock) {.inline.} =
   ## Initializes the given lock.
   when defined(posix):
     var a: SysLockAttr
     initSysLockAttr(a)
     setSysLockType(a, SysLockType_Reentrant)
-    initSysLock(lock, a.addr)
+    initSysLock(lock.lock, a.addr)
   else:
-    initSysLock(lock)
+    initSysLock(lock.lock)
+
+when useOrcArc:
+  proc `=copy`*(x: var RLock, y: RLock) {.error.}
+
+  proc `=destroy`*(lock: var RLock) {.inline.} =
+    deinitSys(lock.lock)
 
-proc deinitRLock*(lock: var RLock) {.inline.} =
-  ## Frees the resources associated with the lock.
-  deinitSys(lock)
+  proc deinitRLock*(lock: var RLock) {.inline, 
+        deprecated: "`deinitRLock` is not needed anymore in ARC/ORC(it is a no-op now); `=destroy` is already defined for `RLock`".} =
```

Comment:
```suggestion
        deprecated: "`deinitRLock` is not needed anymore in ARC/ORC (it is a no-op now); `=destroy` is already defined for `RLock`".} =
```
It would have been easier to just commit my proposed changes ;)